components/shared/components/Widget/components/Widget.tsx

```diff
@@ -216,6 +225,54 @@ export const Widget = withStaticProps(async ({ preview }: { preview: boolean })
     dispatch(fetchReferralsAction.started(undefined));
   }, [dispatch]);
 
+  useEffect(() => {
```

Comment:
Perhaps we should make sure to only prefill once. For instance, if you use open a prefilled widget, change distribution, close and reopen the widget (with the GiveButton), the prefilled distribution will be applied again overwriting any previous changes.

Reply:
Yes, this is a bug! Should run the effect on pre-filled, not the entire widget context. Fixed now and added tests to cover this scenario. If you choose a pre-filled option, change the distribution, and then choose the same pre-fill button, it won't update the distribution, but I think that is a reasonably niche scenario. Might be worth looking out for in hotjar though @EliseAndresen.